src/Compute/Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetDeallocateMethod.cs

```diff
@@ -50,13 +50,27 @@ public override void ExecuteCmdlet()
                     System.Collections.Generic.IList<string> instanceIds = this.InstanceId;
 
                     Rest.Azure.AzureOperationResponse result = null;
-                    if (this.ParameterSetName.Equals("FriendMethod"))
+                    if (NoWait.IsPresent)
                     {
-                        result = VirtualMachineScaleSetsClient.PowerOffWithHttpMessagesAsync(resourceGroupName, vmScaleSetName, instanceIds).GetAwaiter().GetResult();
+                        if (this.ParameterSetName.Equals("FriendMethod"))
+                        {
+                            result = VirtualMachineScaleSetsClient.BeginPowerOffWithHttpMessagesAsync(resourceGroupName, vmScaleSetName, instanceIds).GetAwaiter().GetResult();
+                        }
+                        else
+                        {
+                            result = VirtualMachineScaleSetsClient.BeginDeallocateWithHttpMessagesAsync(resourceGroupName, vmScaleSetName, instanceIds).GetAwaiter().GetResult();
+                        }
                     }
                     else
                     {
-                        result = VirtualMachineScaleSetsClient.DeallocateWithHttpMessagesAsync(resourceGroupName, vmScaleSetName, instanceIds).GetAwaiter().GetResult();
+                        if (this.ParameterSetName.Equals("FriendMethod"))
+                        {
+                            result = VirtualMachineScaleSetsClient.PowerOffWithHttpMessagesAsync(resourceGroupName, vmScaleSetName, instanceIds).GetAwaiter().GetResult();
+                        }
+                        else
+                        {
+                            result = VirtualMachineScaleSetsClient.DeallocateWithHttpMessagesAsync(resourceGroupName, vmScaleSetName, instanceIds).GetAwaiter().GetResult();
+                        }
```

Comment:
same as above, let's revert this for now

Reply:
Done

Reply:
Done